pkg/render/config.go

```diff
@@ -24,7 +24,41 @@ const (
 )
 
 type NetworkConfig struct {
-	CNI                  string
+	CNIPlugin            operatorv1.CNIPluginType
 	NodenameFileOptional bool
 	IPPools              []operatorv1.IPPool
+	MTU                  int32
+	HostPorts            bool
+}
+
+// GenerateRenderConfig converts installation into render config.
+func GenerateRenderConfig(install *operatorv1.Installation) NetworkConfig {
+	networkConfig := NetworkConfig{
+		CNIPlugin: install.Spec.CNI.Type,
+	}
+
+	// Set other provider-specific settings.
+	switch install.Spec.KubernetesProvider {
+	case operatorv1.ProviderDockerEE:
+		networkConfig.NodenameFileOptional = true
+	}
+
+	// If CalicoNetwork is specified, then use Calico networking.
+	if install.Spec.CalicoNetwork == nil {
```

Comment:
I'm confused, the comment says `If CalicoNetwork is specified`, but then the function checks if it's nil...

Reply:
I'm not sure what I was thinking here. I'm removing that if and the associated code block.

Reply:
I found why that was needed. Need to skip the rest of the processing if no CalicoNetwork exists.

Reply:
Ok.  One other concern I have about this function:

Based on the single switch case, it looks like we intend on this function growing as we add more CNI plugins. And it seems a bit fragile that developers need to know whether to add that logic before or after this nil check / return depending on the situation. The comment you've added does help, but worth considering if we can make this less fragile. Not sure what I'd suggest to accomplish that, so I'm fine merging as is and we can re-evaluate as those new cases appear.

Reply:
I could switch the check to `if install.Spec.CalicoNetwork != nil` and then move all the remainder of the code into the code block.